### DISCUSSION TITLE
Anyway to know if a repo/code/project is updated?

### DISCUSSION THREAD

**Discussion by Vitae666** [community/community#22856] [New to GitHub]
<p>What I mean is that without having to go to every single repo/code/project that I use, isn’t there a way to get an email that says, “Hey, this project has been updated!”</p>
<p>There’s times where I’m a few days behind on getting updates because I have no idea if there was one.</p>
<p>What’s the point of Watching and Starring a code/project if you don’t get any info when there’s a change? I’ve checked the settings and there’s nothing there for that.</p>
<p>Sorry, I honestly don’t know what you call these things. Whatever it is that shows when you click “Code” on the page.</p>
<p>I’ve seen Repo, Code and Project thrown around for the same thing, so I don’t know what it’s officially called :slight_smile:</p>

**Comment by lee-dohm** [comment]
<p>It depends on what you mean by “updated”. Watching a repo will notify you any time a new PR is opened and subscribe you to that PR, so you would get a notification when the PR is merged. You can also subscribe to an RSS feed of a repository’s releases, which would notify you of new tags that are pushed to the repository. Unfortunately, if a project is not creating tags or using PRs, then no, there aren’t notifications for “a branch was created” or “commits were pushed” built-in to GitHub. There are some tools that provide that level of notification, for example the <a href="https://github.com/integrations/slack" rel="nofollow noopener">GitHub Slack integration</a> does so, but you have to be an owner or admin of the repository in order to subscribe in that way.</p>
<p>I hope that helps!</p>

**Comment by Vitae666** [comment]
<p>Sorry. I’m knew to the phrases that get used on here, so I’m likely using the wrong terms.</p>
<p>Take this for example:</p>
<p><a href="https://github.com/nls0/C-Hero-Calc" rel="nofollow noopener">https://github.com/nls0/C-Hero-Calc</a></p>
<p>The last version I have is 3.0.2.0g:</p>
<p><a href="https://github.com/nls0/C-Hero-Calc/commit/668c0dfed5e7635c64bf1b7895897ce9cd74f8dd" rel="nofollow noopener">https://github.com/nls0/C-Hero-Calc/commit/668c0dfed5e7635c64bf1b7895897ce9cd74f8dd</a></p>
<p>The current version is 3.0.7.1b</p>
<p><a href="https://github.com/nls0/C-Hero-Calc/commit/f01cc8abd2427f13c80ff9515b551812d4e0b445" rel="nofollow noopener">https://github.com/nls0/C-Hero-Calc/commit/f01cc8abd2427f13c80ff9515b551812d4e0b445</a></p>
<p>I’m 3 updates/commits behind. I don’t recieve emails when this is updated. I’m Watching and Stared it (the main one of course). This is not my repo (or whatever it’s called).</p>
<p>Is there a setting that I am missing that will send me an update when there is an update of ANYTHING on the main page? If 1 out of 20 files is changed, that an email would be sent out?</p>
<p>I’ve gotten emails when Issues are posted.</p>
<p>I’ve gotten emails when the owner has merged a commit that someone else wrote up</p>
<p>As an example, nls0 wrote up a change/commit to something else that I Watch/Star, and when Wiedmolol merged it, I got an email for that.</p>
<p><a href="https://github.com/Wiedmolol/CQMacroCreator/pull/21" rel="nofollow noopener">https://github.com/Wiedmolol/CQMacroCreator/pull/21</a></p>
<p>But I can’t get an email when the owner themselves makes an update? Doesn’t quite make sense to me.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p><a href="https://help.github.com/articles/watching-and-unwatching-repositories/" rel="nofollow noopener">Watching a repository</a>, whether it is yours or someone else’s, will notify you of all new issues and pull requests on that repo. “Repo” is shorthand for “repository”. Watching a repository will not notify you of other activity, such as what you’re collectively calling “updates” to files. Files are changed in repositories only through commits being added to the repository. Commits are added via “branches” and the repository’s “default branch” is what you see on the main page of the repository.</p>
<p>No, there is no setting built-in to GitHub that will notify you of literally any change to a repository or even to only the default branch of the repository. You could set up a tool of some sort via <a href="https://developer.github.com/" rel="nofollow noopener">the GitHub API</a> that would do that for you.</p>
<p>Because the project owner does not use Pull Requests for their own changes to their repo, you are not getting notified of the changes they make. Some project owners use Pull Requests for all changes they make as a historic record of what they did and why they did it. Other project owners use Pull Requests only for “important” changes and only create new commits for “housekeeping” changes like updating documentation or configuration files. It’s all up to people’s preference on how they like to work and how much interaction they want or expect from others.I hope that helps!</p>

**Comment by kientzle** [comment]
<p>Github services used to provide for a reasonable workaround for this omission:  Repo owners could have commit emails sent to a mailing list, then folks could subscribe to the mailing list.</p>
<p>Unfortunately, with the loss of github services, this basic capability is also getting lost.</p>
